csFastFloat/Utils/Utils.cs

```diff
@@ -212,5 +217,57 @@ internal unsafe static bool strncasecmp(byte* input1, byte* input2, int length)
       return (running_diff == 0) || (running_diff == 32);
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static int LeadingZeroCount(ulong value)
+    {
+#if HAS_BITOPERATIONS
+      return System.Numerics.BitOperations.LeadingZeroCount(value);
+#else
+      uint hi = (uint)(value >> 32);
+ 
+      if (hi == 0)
+      {
+        return 32 + Log2SoftwareFallback((uint)value);
+      }
+ 
+      return Log2SoftwareFallback(hi);
+      
+      [MethodImpl(MethodImplOptions.AggressiveInlining)]
+      static int Log2SoftwareFallback(uint value)
+      {
+        // No AggressiveInlining due to large method size
+        // Has conventional contract 0->0 (Log(0) is undefined)
+ 
+        // Fill trailing zeros with ones, eg 00010010 becomes 00011111
+        value |= value >> 01;
+        value |= value >> 02;
+        value |= value >> 04;
+        value |= value >> 08;
+        value |= value >> 16;
+ 
+        // uint.MaxValue >> 27 is always in range [0 - 31] so we use Unsafe.AddByteOffset to avoid bounds check
+        return Unsafe.AddByteOffset(
+          // Using deBruijn sequence, k=2, n=5 (2^5=32) : 0b_0000_0111_1100_0100_1010_1100_1101_1101u
+          ref MemoryMarshal.GetReference(Log2DeBruijn),
+          // uint|long -> IntPtr cast on 32-bit platforms does expensive overflow checks not needed here
+          (IntPtr)(int)((value * 0x07C4ACDDu) >> 27));
+      }
+#endif
+    }
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static unsafe float Int32BitsToSingle(int value)
+    {
+#if HAS_INTRINSICS
```

Comment:
HAS_INTRINSICS is either .NET 5 or .NET Core 3.1 here, so `BitConverter.Int32BitsToSingle` is supported / available and I'd use the method from the framework instead of copying the workaround here.

The workaround got added for .NET 5 (if I've not forgotten when my PR got merged :wink:), so when using `BitConverter.Int32BitsToSingle` .NET Core 3.1 will not emit `vmovd xmm0, ecx` as .NET 5 does and will go to the stack.

IMHO this isn't a huge problem, as .NET 5 is the present and .NET Core 3.1 is the past and we should look into the future.

Besides that for .NET 5 it looks like the workaround isn't needed anymore, and with the pointer-cast the `vmovd xmm0, ecx` is emitted. So I opt for going with the simpler code.

Reply:
It's more complicated that I would expect 😄 Changed to BitConverter